token/token.go

```diff
@@ -0,0 +1,39 @@
+package token
+
+// NOTE:
+// Token data structure is defined below
+// Type (:TokenType: string) => holds the token type to distinguish between strings, numbers, right brackets, semi-colons etc.
+// Literal (:string) => holds the literal string value of the token
+
+type TokenType string
+
+type Token struct {
+	Type    TokenType
+	Literal string
+}
+
+// NOTE:
+// Below are all valid lexer tokens for JSON
+const (
```

Comment:
Add constant values for identifiers. Find out if these can be only string in JSON